### ISSUE TITLE
Potenial Widescreen Issue (Good example for issue request)

### ISSUE THREAD

**Issue by lds1998** [kbdevs/ai-aimbot#32]
Hi, when in use the aim goes off-center and loses the target. At the same time, the aimbot doesn't aim properly at the targets. I have annexed a video file and 2 images of the happening behavior.
Info:
Game: Cyberpunk 2077 v2.01
GPU: AMD Radeon RX 6950 XT
CPU: AMD Ryzen 7 5800X3D
Windows Resolution: 3440 x 1440 @165hz
Game Resolution: 3440 x 1440 @165hz
OS:  Microsoft Windows 11 Professional (x64) Build 22621.2361 (22H2)

It was tested on both versions v1.3 and v1.2 with the same behavior.

![Before Aim-min](https://github.com/kbdevs/ai-aimbot/assets/24820190/7980071e-4031-4bc6-8084-4a696685e147)
![After Aim-min](https://github.com/kbdevs/ai-aimbot/assets/24820190/9d81be67-0b9d-40cf-b619-829e1587006d)

https://github.com/kbdevs/ai-aimbot/assets/24820190/7401721c-4e60-40cd-b25c-16485dd02f8a

**Comment by kbdevs** [comment]
First of all thanks so much for all the info at the start.

Im pretty sure this is due to it trying to grab a center 320px window cut from a 1920x1080 frame, if you could check to see if it works in 1920x1080 for you that would be appreciated, if that is the issue ill look at fixing the programs window grabbing, if that isnt we'll try more things.

**Comment by lds1998** [comment]
So I did some testing and Cyberpunk doesn't allow to pin the windows on top of it, tried deskpins and powertoys.

To record the behavior I changed the Desktop resolution to 2560x1440 and the In-game resolution to 1920x1080 and have annexed here the capture ( Test 1440, 1080.mp4).

I also tried 1920x1080 resolution for both Desktop and In-game and the same behavior happened. ( Test 1080.mp4).

Did also tested the Desktop resolution of 1920x1080 and the In-game resolution of 1280x720, same behavior.



https://github.com/kbdevs/ai-aimbot/assets/24820190/b31ffe23-457d-4f88-ad20-4cac0a457ef6



https://github.com/kbdevs/ai-aimbot/assets/24820190/51a4b4c6-e167-4571-ac94-3168e820251b

**Comment by kbdevs** [comment]
In test.1080p.mp4 i think the crosshair is making it lose the current locked enemy, you could try changing the confidence in this line (change 0.7 to a decimal 0 to 1, havent worked on that bit for a while so try increase and decrease the decimal and see if any difference)

if confidence > 0.7 and classID == 0:

The 1440,1080 one is detecting wrong likely due to res mismatch, you can see in the video that the detection window is in the wrong spot

and for the 1080p.mp4 was it in windowed fullscreen?

**Comment by lds1998** [comment]
Yes, i tested the game both in window and borderless window modes but couldn't pin.
I am going try using obs for the recording to see if i can stitch it better.
I will try updating the confidence to see if it works.

**Comment by kbdevs** [comment]
Hmm I think this just might not be a great fit for cyberpunk, I will look into fixing the widescreen issue.

**Comment by lds1998** [comment]
I used Cyberpunk because is the only offline game I have that has humans on it and I could show the issue without too many variables.
The problem is the same even in multiplayer games like Battlefield and Warzone, tested as I did Cyberpunk.

I tried to convert your project to be able to run DirectML or [MIVisionX](https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX), so I could have GPU acceleration for my AMD GPU like Nvidia Cuda but can't manage to convert your settings to them without terrible loss of performance or breaking the entire thing, so I did the next best thing and got 1060 to test it out to see if the issue maybe be caused because of the CPU fallback. Gonna test it and keep you posted.

**Comment by Bazz0r** [comment]
@lds1998 sorry to hijack the thread, did you try it on Warzone? I found it does not move the mouse, did you encounter the same? Thank you.

**Comment by lds1998** [comment]
@Bazz0r I tried it in Warzone and it kinda works but only after changing the output of the Bot to a hacked mouse driver to bypass the anti-cheat, same thing that must be done for Valorant btw, but didn't do more testing since I can't get the aimbot to properly work and I am trying to find a way to make aimbot work with the dev's help.
So my info is a bit incomplete and can't provide a more detailed guide until I can get the main part to work, but the basic of it is:

There are a couple of ways to bypass it:

-Hacked mouse drivers or Hacked controller drivers that are signed, are shared on the internet forums but can be dangerous because they can have malicious code in them if you grab them from a malicious source;

- Use remote play using a console like Playstation or Xbox, since those programs don't have anti-cheat on the remote play but may require changing the bot outputs to match the controller outputs in certain cases to get it working correctly.

-Use of an Arduino or similar micro-programmable computer that receives the commands of the bot from the computer and then sends them back to the computer on another USB HID device. ( this one works for games running locally, on a console, another PC, and even a smartphone but requires extra hardware to work but is super compatible )

I would rank them: ( Best<...<Worst)
By ranking them from easy to hard: Remote play < Hacked Driver < Arduino
By ranking them from security points both personal and anti-cheat levels: Arduino < Remote Play < Hacked Driver

**Comment by Bazz0r** [comment]
@lds1998 thank you, I appreciate the response!

**Comment by kbdevs** [comment]
Hey @lds1998 where did you get the hacked mouse driver?

**Comment by lds1998** [comment]
If you want a premade one here's the one I got working with the bot: [1st](https://www.unknowncheats.me/forum/valorant/598337-mouse-movement-driver.html), [2nd](https://github.com/ddxoft/master) (You need chatgpt to translate to English to understand the functions)

If you want to make your own driver (not signed): [Tutorial](https://www.unknowncheats.me/forum/anti-cheat-bypass/271733-driver-aka-kernel-mode.html)

If you got an Arduino: [Github](https://github.com/arduino-libraries/Mouse)

Both the 1st and Arduino solutions were tested by me to be working after being implemented on the aimbot code but they work really badly and need to be better implemented and optimized the reason for the issues is simple because I am a bad coder :P
